examples/multi-engine-setup/js/button-start-ar-session.ts

```diff
@@ -0,0 +1,47 @@
+/**
+ * ButtonStartARSession
+ * Waits until the ARSession is ready and shows the default AR button.
+ * AR button, when clicked, searches for the first component which is `instanceOf ARCamera`
+ * and starts that components AR session
+ */
+import {Component} from '@wonderlandengine/api';
+import {ARSession, ARCamera} from '@wonderlandengine/8thwall-tracking';
+
+export class ButtonStartARSession extends Component {
+    public static TypeName = 'button-start-ar-session';
+
+    init() {
+        if (ARSession.getSessionForEngine(this.engine).arSessionReady) {
+            this.onARSessionReady();
+        } else {
+            ARSession.getSessionForEngine(this.engine).onARSessionReady.add(this.onARSessionReady);
+        }
```

Comment:
We now have `RetainEmitter` for this, which will immediately run the callback, if there is an event retained (until reset).

So this would become `ARSession.getSessionForEngine(this.engine).onARSessionReady.add(this.onARSessionReady);` only, but make sure to call `onARSessionReady.reset()` on session end.

Since there is the semantic context of "ARSession" already, you could probably call it `onSessionReady` instead btw, but maybe that just leads to confusion with onXRSessionStart 🤔